src/PowerShellEditorServices/Debugging/DebugService.cs

```diff
@@ -709,16 +785,24 @@ public VariableScope[] GetVariableScopes(int stackFrameId)
 
         private async Task FetchStackFramesAndVariables(string scriptNameOverride)
         {
-            this.nextVariableId = VariableDetailsBase.FirstVariableId;
-            this.variables = new List<VariableDetailsBase>();
+            await this.stackFramesHandle.WaitAsync();
+            try
+            {
+                this.nextVariableId = VariableDetailsBase.FirstVariableId;
+                this.variables = new List<VariableDetailsBase>();
 
-            // Create a dummy variable for index 0, should never see this.
-            this.variables.Add(new VariableDetails("Dummy", null));
+                // Create a dummy variable for index 0, should never see this.
+                this.variables.Add(new VariableDetails("Dummy", null));
```

Comment:
Why is this needed? This looks like a code smell if the code cannot cope with an empty list.